tests/src/providers/testqgscopcprovider.cpp

```diff
@@ -191,15 +194,16 @@ void TestQgsCopcProvider::querySublayers()
   QCOMPARE( res.at( 0 ).type(), QgsMapLayerType::PointCloudLayer );
 
   // make sure result is valid to load layer from
-  const QgsProviderSublayerDetails::LayerOptions options{ QgsCoordinateTransformContext() };
+  QgsProviderSublayerDetails::LayerOptions options{ QgsCoordinateTransformContext() };
+  options.skipStatisticsCalculation = true;
   std::unique_ptr< QgsPointCloudLayer > ml( qgis::down_cast< QgsPointCloudLayer * >( res.at( 0 ).toLayer( options ) ) );
   QVERIFY( ml->isValid() );
 }
 
 void TestQgsCopcProvider::brokenPath()
 {
   // test loading a bad layer URI
-  std::unique_ptr< QgsPointCloudLayer > layer = std::make_unique< QgsPointCloudLayer >( QStringLiteral( "not valid" ), QStringLiteral( "layer" ), QStringLiteral( "copc" ) );
+  std::unique_ptr< QgsPointCloudLayer > layer = std::make_unique< QgsPointCloudLayer >( QStringLiteral( "not valid" ), QStringLiteral( "layer" ), QStringLiteral( "copc" ), mNoStatsOption );
```

Comment:
Why do we need to change unit test here and in other tests? the behavior should not be changed... or am I missing something?

Reply:
The layer will start a task for calculating the statistics and in unit tests the task might finish after the layer is destroyed and things will crash at that point, the tests where failing because of that so I nneded to add the option to skip the statistics calculation...